servers/zms/src/main/java/com/yahoo/athenz/zms/ZMSImpl.java

```diff
@@ -2681,6 +2683,37 @@ public AuthHistoryDependencies getAuthHistoryDependencies(ResourceContext ctx, S
         return dbService.getAuthHistory(domainName);
     }
 
+    @Override
+    public void deleteExpiredMembers(ResourceContext ctx, Integer purgeResources) {
+        final String caller = ctx.getApiName();
+        logPrincipal(ctx);
+        validateRequest(ctx.request(), caller);
+
+        LOG.info("deleteExpiredMembers: purgeResources value: {}", purgeResources);
+
+        if (purgeResources == null) {
+            // it is not provided, all resources will be purged
+            purgeResources = 3;
+        }
+
+        if (!List.of(0, 1, 2, 3).contains(purgeResources)) {
+            throw ZMSUtils.requestError("Invalid purgeResources value, error: should be a number in [0-3]", caller);
+        }
+
+        EnumSet<PurgeResourcesEnum>  purgeResourcesEnumSet = PurgeResourcesEnum.getPurgeResourcesState(purgeResources);
+
+        LOG.info("deleteExpiredMembers: purgeResources: {}", purgeResourcesEnumSet);
+
+        if (purgeResourcesEnumSet.contains(PurgeResourcesEnum.ROLES)) {
```

Comment:
a follow up from the comment in the RDL, let's collect and return the value if requested